src/command_modules/azure-cli-appservice/azure/cli/command_modules/appservice/custom.py

```diff
@@ -949,3 +950,58 @@ def _match_host_names_from_cert(hostnames_from_cert, hostnames_in_webapp):
         elif hostname in hostnames_in_webapp:
             matched.add(hostname)
     return matched
+
+
+def create_function(resource_group_name, name, storage_account, plan=None):
+    client = web_client_factory()
+    if is_valid_resource_id(plan):
+        plan = parse_resource_id(plan)['name']
+    location = _get_location_from_app_service_plan(client, resource_group_name, plan)
+    webapp_def = Site(server_farm_id=plan, location=location)
+    webapp_def.kind = 'functionapp'
+    # check for storage account
+    sa_resource_group = resource_group_name
+    if storage_account.startswith('/subscriptions/'):
```

Comment:
You really don't even need `is_vaild_resource_id`. If you call `parse_resource_id('name_only')` you'll get a dictionary where name = 'name_only' and the other parts are None.